src/backend/executor/index_scan_executor.cpp

```diff
@@ -179,13 +179,21 @@ bool IndexScanExecutor::ExecIndexLookup() {
         // perform predicate evaluation.
         if (predicate_ == nullptr) {
           visible_tuples[tile_group_id].push_back(tuple_id);
-          transaction_manager.PerformRead(tile_group_id, tuple_id);
+          auto res = transaction_manager.PerformRead(tile_group_id, tuple_id);
+          if(!res){
+            transaction_manager.SetTransactionResult(RESULT_FAILURE);
+            return res;
+          }
         } else {
           expression::ContainerTuple<storage::TileGroup> tuple(tile_group.get(), tuple_id);
           auto eval = predicate_->Evaluate(&tuple, nullptr, executor_context_).IsTrue();
           if (eval == true) {
             visible_tuples[tile_group_id].push_back(tuple_id);
-            transaction_manager.PerformRead(tile_group_id, tuple_id);
+            auto res = transaction_manager.PerformRead(tile_group_id, tuple_id);
+            if(!res){
```

Comment:
you mean SetOutput? I think it is unnecessary to do that in this case, as the transaction fails, and no output should be returned.